bindings/java/src/integration/com/apple/foundationdb/DirectoryTest.java

```diff
@@ -0,0 +1,233 @@
+
+                    path.add("bar");
+                    correctPath = correctPath.resolve("bar");
+                    DirectorySubspace bar = dir.create(tr,path).join();
+                    Assertions.assertEquals(correctPath, toPathObject(bar.getPath()),"incorrect path");
+                    Assertions.assertTrue( bar.exists(tr).join(),"does not exist even though it's been created!");
+                } finally {
+                    // remove the directory
+                    dir.remove(tr, path).join();
+                }
+                return null;
+            });
+        }
+    }
+
+    @Test
+    public void testCanMoveSubDirectory() throws Exception {
+        /*
+         * Make sure that we can move a subdirectory correctly
+         */
+        final DirectoryLayer dir = new DirectoryLayer();
+        try (Database db = fdb.open()) {
+
+            db.run(tr -> {
+                List<String> path = new ArrayList<>();
+                path.add("src");
+                Path correctPath = Paths.get("src");
+                try {
+                    DirectorySubspace foo = dir.create(tr, path).join();
+                    Assertions.assertEquals( correctPath, toPathObject(foo.getPath()),"Incorrect path");
+                    // make sure it exists
+                    Assertions.assertTrue( foo.exists(tr).join(),"does not exist even though it's been created!");
+
+                    path.add("bar");
+                    correctPath = correctPath.resolve("bar");
+                    DirectorySubspace bar = dir.create(tr,path).join();
+                    Assertions.assertEquals(correctPath, toPathObject(bar.getPath()),"incorrect path");
+                    Assertions.assertTrue( bar.exists(tr).join(),"does not exist even though it's been created!");
+
+                    DirectorySubspace boo = dir.create(tr,Arrays.asList("dest")).join();
+                    Assertions.assertEquals(Paths.get("dest"), toPathObject(boo.getPath()),"incorrect path");
+                    Assertions.assertTrue( boo.exists(tr).join(),"does not exist even though it's been created!");
+
+                    //move the subdirectory and see if it works
+                    DirectorySubspace newBar = bar.moveTo(tr,Arrays.asList("dest","bar")).join();
+                    Assertions.assertEquals(Paths.get("dest","bar"), toPathObject(newBar.getPath()),"incorrect path");
+                    Assertions.assertTrue( newBar.exists(tr).join(),"does not exist even though it's been created!");
+                    Assertions.assertFalse(bar.exists(tr).join(),"Still exists in old location!");
+
+                } finally {
+                    // remove the directory
+                    dir.remove(tr, Arrays.asList("src")).join();
+                    try{
+                        dir.remove(tr, Arrays.asList("dest")).join();
+                    }catch(CompletionException ce){
```

Comment:
```suggestion
                    } catch (CompletionException ce){
```
I am not sure what tool you are using, Intellij should autoformat this for you.